ExampleMod/Old/ExampleMod.cs

```diff
@@ -144,12 +140,6 @@ public class ExampleMod : Mod
 		}
 
 		public override void Unload() {
-			// All code below runs only if we're not loading on a server
-			if (!Main.dedServ) {
-				Main.tileFrame[TileID.Loom] = 0; // Reset the frame of the loom tile
-				Main.tileSetsLoaded[TileID.Loom] = false; // Causes the loom tile to reload its vanilla texture
```

Comment:
Is there a reason you abandoned this approach? It seems superior to me.

Reply:
1.3:
```cs
public void LoadTiles(int i) {
	if (!tileSetsLoaded[i]) {
		tileTexture[i] = OurLoad<Texture2D>("Images" + Path.DirectorySeparatorChar.ToString() + "Tiles_" + i);
		tileSetsLoaded[i] = true;
	}
}```

1.4:
```cs
public void LoadTiles(int i) {
	if (TextureAssets.Tile[i].State == AssetState.NotLoaded)
		Assets.Request<Texture2D>(TextureAssets.Tile[i].Name);
}```

Yes, the 1.3 design is superior, I agree. `tileSetsLoaded` doesn't exist in 1.4 anymore, the new loading is more of a "request", the actual loading happens much earlier in AssetInitializer:
`TextureAssets.Tile[n] = LoadAsset<Texture2D>("Images/Tiles_" + n, AssetRequestMode.DoNotLoad);`

So yeah swapping made the most sense to me at the time of writing the PR because it has to go by the assets name to actually "finalize" the texture used. There's also no way to force an asset to be unloaded so it can enter the NotLoaded state (setter private, some methods internal)

Reply:
Also the intention of this PR is to make tml devs think about the Asset system more :P